plugins/org.jkiss.dbeaver.model.sql/src/org/jkiss/dbeaver/model/sql/translate/SQLQueryTranslator.java

```diff
@@ -14,47 +14,182 @@
+        if (statement != null) {
+            return translateStatement(query, statement);
         }
-        return sql.toString();
+        return Collections.singletonList(query);
     }
 
+    /**
+     * Translates statement to target dialect.
+     *
+     * @param query     the query
+     * @param statement the statement
+     * @return the list
+     */
+    protected List<? extends SQLScriptElement> translateStatement(@NotNull SQLQuery query,
+                                                                  @NotNull Statement statement) {
+
+        List<SQLScriptElement> extraQueries = null;
+
+        if (statement instanceof CreateTable) {
+            boolean defChanged = false;
+            CreateTable createTable = (CreateTable) statement;
+            SQLDialect targetDialect = sqlTranslateContext.getTargetDialect();
+            SQLDialectDDLExtension extendedDialect = null;
+            if (targetDialect instanceof SQLDialectDDLExtension) {
+                extendedDialect = (SQLDialectDDLExtension) targetDialect;
+            }
+
+            if (extendedDialect != null && extendedDialect.supportsCreateIfExists()) {
+                createTable.setIfNotExists(false);
+                defChanged = true;
+            }
+
+            if (!targetDialect.supportsTableDropCascade() && createTable.getIndexes() != null) {
+                for (Index index : createTable.getIndexes()) {
+                    if (index instanceof ForeignKeyIndex) {
+                        ForeignKeyIndex foreignKeyIndex = (ForeignKeyIndex) index;
+                        ReferentialAction referentialAction = foreignKeyIndex.getReferentialAction(ReferentialAction.Type.DELETE);
+                        if (referentialAction != null && referentialAction.getAction().equals(ReferentialAction.Action.CASCADE)) {
+                            referentialAction.setAction(ReferentialAction.Action.NO_ACTION);
+                            defChanged = true;
+                        }
+                    }
+                }
+            }
+
+            for (ColumnDefinition cd : createTable.getColumnDefinitions()) {
+                String newDataType = null;
+                switch (cd.getColDataType().getDataType().toUpperCase(Locale.ENGLISH)) {
+                    case "CLOB":
+                    case "TEXT":
+                        newDataType = (extendedDialect != null) ? extendedDialect.getLargeCharacterType() : "varchar";
+                        break;
+                    case "TIMESTAMP":
+                        if (extendedDialect != null && extendedDialect.timestampAsDatetime()) {
+                            newDataType = "datetime";
+                        }
+                        break;
+                    case "BIGINT":
+                        if (extendedDialect != null) {
+                            newDataType = extendedDialect.getLargeNumericType();
+                        }
+                        break;
+                }
```

Comment:
Please add an empty `default` branch to make it clear that no action should be performed.